### DISCUSSION TITLE
SHA in dependency resolution?

### DISCUSSION THREAD

**Discussion by kleinsch** [pnpm/pnpm#6070] [Q&A] [answered]
I have dependencies resolving with what look like SHAs following an underscore, so they get resolved separately: `'@chakra-ui/react': 1.8.9_9181b4d5ce4999e50bf2497382e51d0f`

I know peer dependencies need to be resolved separately, and I see that happen for other things in the repo: `1.2.3_react@17.0.2`

What are the SHAs? What are they referring to?

**Comment by gluxon** [comment, marked as answer]
When the `peersSuffix` becomes too long, pnpm hashes the string to keep it short due to Windows file system path length limitations. Zoltan improved this in v8, so the hashes will be readable again.

https://twitter.com/pnpmjs/status/1613188416572628992?cxt=HHwWgICx2auUmeMsAAAA

If you're curious what `9181b4d5ce4999e50bf2497382e51d0f` refers to on pnpm v7 it should be possible to compare the `peerDependencies` block with the `dependencies` block in the lockfile and see what peers were met. If a peer dependency is missing, it won't be present in the `dependencies` block of `@chakra-ui/react/1.8.9_9181b4d5ce4999e50bf2497382e51d0f`.

**Reply by kleinsch** [reply]
Makes sense, thanks!

**Reply by zkochan** [reply]
You can use the new lockfile format in v7 as well. Just set `use-lockfile-v6=true`.
